src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx

```diff
@@ -264,28 +288,28 @@ const SatelliteManifestPanel: FunctionComponent<SatelliteManifestPanelProps> = (
     return rowsWithAllocationDetails;
   };
 
-  const toggleAllocationDetails = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
-    const rowIndexToUpdate = rowKey / 2;
-    const newRowExpandedStatus = [...rowExpandedStatus];
-    newRowExpandedStatus[rowIndexToUpdate] = isOpen;
-    setRowExpandedStatus(newRowExpandedStatus);
+  const toggleAllocationDetails = (
+    event: React.MouseEvent,
+    rowKey: number,
+    isOpen: boolean,
+    rowData: any
+  ) => {
+    const uuid: string = rowData.uuid.title;
+    rowExpandedStatus[uuid] = !rowExpandedStatus[uuid];
```

Comment:
A couple things here - first, as mentioned above, the forceUpdate should be removed. 

Second, you shouldn't modify state directly but only through the setState method (here setRowExpandedStatus). 

Objects can be a bit tricky bc Javascript passes objects by reference and not by value. So if you did 

`let newRowExpandedStatus = rowExpandedStatus` 

it would just be a new pointer to the same old object. 

You should copy the object and then modify it and then use the setRowExpandedStatus to the newRowExpandedStatus. Something like this: 

```
const newRowExpandedStatus = { ...rowExpandedStatus };
newRowExpandedStatus[uuid] = !rowExpandedStatus[uuid];
setRowExpandedStatus(newRowExpandedStatus);
```
The spread operator (...) works here; you can also use Object.assign({}, rowExpandedStatus) but it's a bit clunkier. 

Then the setRowExpandedStatus updates the state, and React rerenders the component.

Reply:
I didn't like needing to force the update, but I completely forgot about this crazy spread pattern.  I saw it explained in a tutorial somewhere, sometime.  It's so counterintuitive and [trivially] inefficient to recreate an object from scratch to change one key value pair; kind of defeats the purpose of such an object.  I'll update.